### PR TITLE
[qfix] Delete from clientconn map if dial fails

### PR DESCRIPTION
**Pull request by glazychev-art**
Signed-off-by: Artem Glazychev <artem.glazychev@xored.com>

## Motivation
Fixes https://github.com/networkservicemesh/sdk/files/7519966/nsmgr-phjrv-previous.txt

## Description
If dial fails we need to delete `dialer` from metadata

## Types of changes

- [x] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/networkservice/common/dial/dialer.go [hunk 1]
```diff
@@ -72,6 +73,7 @@ func (di *dialer) Dial(ctx context.Context, clientURL *url.URL) error {
 		if cc != nil {
 			_ = cc.Close()
 		}
+		log.FromContext(ctx).Errorf("failed to dial %s", target)
```
Comment (denis-tingaikin) [inline pkg/networkservice/common/dial/dialer.go:76]:
```suggestion
```

File: pkg/networkservice/common/dial/client.go [hunk 2]
```diff
@@ -74,12 +74,17 @@ func (d *dialClient) Request(ctx context.Context, request *networkservice.Networ
...
+		if err != nil {
+			clientconn.Delete(ctx)
+			return nil, err
+		}
 		_, _ = next.Client(ctx).Close(clienturlctx.WithClientURL(closeCtx, di.clientURL), request.GetConnection(), opts...)
 	}
 
 	err := di.Dial(ctx, clientURL)
 	if err != nil {
+		clientconn.Delete(ctx)
```
Comment (denis-tingaikin) [inline pkg/networkservice/common/dial/client.go:87]:
```suggestion

		log.FromContext(ctx).Errorf("can not dial to %v, err %v. Deleting clientconn...", target, err)
		clientconn.Delete(ctx)
```

File: pkg/networkservice/common/dial/client.go [hunk 3]
```diff
@@ -74,12 +74,17 @@ func (d *dialClient) Request(ctx context.Context, request *networkservice.Networ
 	if di.clientURL != nil && di.clientURL.String() != clientURL.String() {
 		closeCtx, closeCancel := closeContextFunc()
 		defer closeCancel()
-		_ = di.Dial(closeCtx, di.clientURL)
+		err := di.Dial(closeCtx, di.clientURL)
+		if err != nil {
+			clientconn.Delete(ctx)
```
Comment (denis-tingaikin) [inline pkg/networkservice/common/dial/client.go:79]:
```suggestion

		log.FromContext(ctx).Errorf("can not redial to %v, err %v. Deleting clientconn...", target, err)
		clientconn.Delete(ctx)
```

**Review by denis-tingaikin — COMMENTED** [review]
LGTM

File: pkg/networkservice/common/dial/client.go (hunk 2 above)
Comment (glazychev-art) [reply pkg/networkservice/common/dial/client.go:91]: done

**Review by denis-tingaikin — APPROVED** [review]
@edwarnicke we found this issue on CI.


[nsmgr-phjrv-previous.txt](https://github.com/networkservicemesh/sdk/files/7519966/nsmgr-phjrv-previous.txt)


So merging

**Pull request merged**